source/MaterialXRenderGlsl/TextureBaker.cpp

```diff
@@ -180,62 +199,93 @@ void TextureBaker::writeBakedMaterial(const FilePath& filename, const StringVec&
     DocumentPtr bakedTextureDoc = createDocument();
 
     // Create top-level elements.
-    NodeGraphPtr bakedNodeGraph = bakedTextureDoc->addNodeGraph("NG_baked");
-    GeomInfoPtr bakedGeom = !udimSet.empty() ? bakedTextureDoc->addGeomInfo("GI_baked") : nullptr;
+    const string bakedNodeGraphName = bakedTextureDoc->createValidChildName("NG_baked");
```

Comment:
Found that you could have two nodes of the same name being created so fixed this.